tests/test_state/test_storage_state.py

```diff
@@ -36,6 +57,128 @@ def test_market_cycle():
+        storage_state.market_cycle(past_time_slot, current_time_slot, future_time_slots)
+        assert storage_state.used_storage == 90
+        storage_state.pledged_sell_kWh[past_time_slot] = 0
+        storage_state.pledged_buy_kWh[past_time_slot] = 10
+        storage_state.market_cycle(past_time_slot, current_time_slot, future_time_slots)
+        assert storage_state.used_storage == 100
+
+    def test_market_cycle_ess_share_time_series_dict(self):
+        storage_state = StorageState(initial_soc=100,
+                                     capacity=100,
+                                     initial_energy_origin=ESSEnergyOrigin.LOCAL,
+                                     max_abs_battery_power_kW=15)
+        past_time_slot, current_time_slot, future_time_slots = self._initialize_time_slots()
+        active_market_slot_time_list = [past_time_slot, current_time_slot, *future_time_slots]
+        storage_state.add_default_values_to_state_profiles(active_market_slot_time_list)
+        energy = 0.3
+        storage_state.activate(
+            slot_length=duration(minutes=15), current_time_slot=current_time_slot)
+        storage_state.register_energy_from_one_sided_market_accept_offer(
+            energy, past_time_slot, ESSEnergyOrigin.LOCAL)
+        storage_state.register_energy_from_one_sided_market_accept_offer(
+            energy, past_time_slot, ESSEnergyOrigin.UNKNOWN)
+        storage_state.register_energy_from_one_sided_market_accept_offer(
+            energy, past_time_slot, ESSEnergyOrigin.UNKNOWN)
+        storage_state.market_cycle(past_time_slot, current_time_slot, future_time_slots)
+        expected_time_series = {ESSEnergyOrigin.LOCAL: storage_state.initial_capacity_kWh + energy,
+                                ESSEnergyOrigin.EXTERNAL: 0.0,
+                                ESSEnergyOrigin.UNKNOWN: 2 * energy}
+        assert storage_state.time_series_ess_share[past_time_slot] == expected_time_series
+
+    def test_market_cycle_calculate_and_update_soc_and_set_offer_history(self):
+        storage_state = StorageState(initial_soc=100,
+                                     capacity=100,
+                                     min_allowed_soc=20)
+        past_time_slot, current_time_slot, future_time_slots = self._initialize_time_slots()
+        active_market_slot_time_list = [past_time_slot, current_time_slot, *future_time_slots]
+        storage_state.add_default_values_to_state_profiles(active_market_slot_time_list)
+        storage_state.activate(
+            slot_length=duration(minutes=15), current_time_slot=current_time_slot)
+        storage_state.pledged_sell_kWh[past_time_slot] = 10
+        storage_state.market_cycle(past_time_slot, current_time_slot, future_time_slots)
+        assert storage_state.charge_history[current_time_slot] != storage_state.initial_soc
+        assert storage_state.charge_history_kWh[current_time_slot] != \
+               storage_state.initial_capacity_kWh
```

Comment:
Please avoid backslashes. Thanks!
```suggestion
        assert (storage_state.charge_history_kWh[current_time_slot] != 
               storage_state.initial_capacity_kWh)
```